orc8r/gateway/python/magma/common/sentry.py

```diff
@@ -83,30 +85,55 @@ def get_sentry_status(service_name: str) -> SentryStatus:
         return SentryStatus.DISABLED
 
 
-def sentry_init(service_name: str):
-    """Initialize connection and start piping errors to sentry.io."""
+# TODO when control_proxy.yml is outdated move to shared mconfig entirely
+def get_sentry_dsn_and_sample_rate(sentry_mconfig: mconfigs_pb2.SharedSentryConfig) -> Tuple[str, float]:
+    """Get Sentry configs with the following priority
 
-    sentry_status = get_sentry_status(service_name)
-    if sentry_status == SentryStatus.DISABLED:
-        return
+    1) control_proxy.yml (if sentry_python_url is present)
+    2) shared mconfig (i.e. first streamed mconfig from orc8r,
+    if not present default mconfig in /etc/magma)
 
-    sentry_url = get_service_config_value(
+    :param sentry_mconfig: proto message type of shared mconfig
+    :return: (str, float) with sentry url, sentry sample rate
+    """
+    dsn_python = get_service_config_value(
         CONTROL_PROXY,
         SENTRY_URL,
         default='',
     )
-    if not sentry_url:
-        return
 
-    sentry_sample_rate = get_service_config_value(
+    if not dsn_python:
+        dsn_python = sentry_mconfig.dsn_python
+        sample_rate = sentry_mconfig.sample_rate
+        return dsn_python, sample_rate
+
+    sample_rate = get_service_config_value(
         CONTROL_PROXY,
         SENTRY_SAMPLE_RATE,
-        default=1.0,
+        default=DEFAULT_SAMPLE_RATE,
     )
+    return dsn_python, sample_rate
+
+
+def sentry_init(service_name: str, sentry_mconfig: mconfigs_pb2.SharedSentryConfig) -> None:
+    """Initialize connection and start piping errors to sentry.io."""
+
+    sentry_status = get_sentry_status(service_name)
+    if sentry_status == SentryStatus.DISABLED:
+        return
+
+    dsn_python, sample_rate = get_sentry_dsn_and_sample_rate(sentry_mconfig)
+
+    if not dsn_python:
+        logging.info('Sentry disabled because of missing dsn_python. '
+                     'See documentation (Configure > AGW) on how to configure '
```

Comment:
👍

Reply:
Pushed one more time to make format check happy (and also markdown lint check seems to be broken on master. ran make precommit_fix in docs and included changes here).